packages/sdk/src/types/Qualifications.ts

```diff
@@ -0,0 +1,109 @@
+import { CustomerRequest } from './Customers'
+import { Junction } from './Exports'
+import { OrderRequest, OrderResponse } from './Orders'
+import { ApplicableToResultList, InapplicableToResultList } from './ApplicableTo'
+import { Category } from './Categories'
+import { Discount } from './DiscountVoucher'
+import { LoyaltiesTransferPoints } from './Loyalties'
+import { ValidationRulesAssignmentsList } from './ValidationRules'
+
+export type QualificationsCheckEligibilityRequestBody = {
+	customer?: CustomerRequest
+	order?: OrderRequest
+	mode?: 'BASIC' | 'ADVANCED'
+	tracking_id?: string
+	scenario?:
+		| 'ALL'
+		| 'CUSTOMER_WALLET'
+		| 'AUDIENCE_ONLY'
+		| 'PRODUCTS'
+		| 'PRODUCTS_DISCOUNT'
+		| 'PROMOTION_STACKS'
+		| 'PRODUCTS_BY_CUSTOMER'
+		| 'PRODUCTS_DISCOUNT_BY_CUSTOMER' // default ALL
+	options?: {
+		limit?: number //max 100
+		starting_after?: string
+		filters?: {
+			junction?: Junction
+		} & Partial<Record<QualificationsFiltersFields, QualificationsFieldConditions>>
+		expand?: ('redeemable' | 'category' | 'validation_rules')[]
+		sorting_rule?: 'BEST_DEAL' | 'LEAST_DEAL' | 'DEFAULT'
+	}
+	metadata?: Record<string, any>
+}
+
+export type QualificationsCheckEligibilityResponseBody = {
+	redeemables: QualificationsRedeemablesResponse
+	tracking_id?: string
+	order?: OrderResponse
+	stacking_rules: QualificationsStackingRulesResponse
+}
+
+//domain type
+export type QualificationsFiltersFields =
+	| 'category_id'
+	| 'campaign_id'
+	| 'resource_id'
+	| 'resource_type'
+	| 'voucher_type'
+	| 'code'
+
+export type QualificationsFiltersCondition = '$in' | '$not_in' | '$is' | '$is_not' | '$has_value' | '$is_unknown'
+
+export type QualificationsFieldConditions = { conditions?: Partial<Record<QualificationsFiltersCondition, any>> }
+
+export type QualificationsRedeemablesResponse = {
+	object: 'list'
+	data_ref: 'data'
+	data: QualificationsRedeemableSingleResponse[]
+	total: number
+	has_more: boolean
+	more_starting_after?: string
+}
+
+export type QualificationsStackingRulesResponse = {
+	redeemables_limit: number
+	applicable_redeemables_limit: number
+	applicable_exclusive_redeemables_limit: number
+	exclusive_categories: string[]
+	joint_categories: string[]
+}
+
+export type QualificationsRedeemableSingleResponse = QualificationsRedeemableSingleResponseBase & {
+	redeemables?: QualificationsRedeemableSingleResponseBase[]
```

Comment:
so in this case we will get a repeated `QualificationsRedeemableSingleResponseBase` type in redeemables? I'm asking because I couldn't get the result with the `redeemables` even though I sent  `"options": {
"expand": ["redeemable"] }` and it seems strange to me 😅

Reply:
and the doc shows that it should return redeemables' metadata (?)
<img width="521" alt="image" src="https://github.com/voucherifyio/voucherify-js-sdk/assets/117282008/a67f5e63-9259-4b17-9d34-6f24b54af0c9">

Reply:
```
{
    "redeemables": {
        "object": "list",
        "data_ref": "data",
        "data": [
            {
                "id": "string",
                "object": "campaign",
                "created_at": "2022-03-09T11:19:04.819Z",
                "result": {
                    "discount": {
                        "type": "AMOUNT",
                        "amount_off": 0,
                        "amount_off_formula": "string",
                        "aggregated_amount_limit": 0,
                        "effect": "APPLY_TO_ORDER",
                        "is_dynamic": true
                    },
                    "gift": {
                        "balance": 0,
                        "credits": 0
                    },
                    "loyalty_card": {
                        "points": 7000,
                        "balance": 6970,
                        "exchange_ratio": 0,
                        "points_ratio": 0,
                        "transfers": [
                            {
                                "code": "string",
                                "points": 0,
                                "reason": "string",
                                "source_id": "string"
                            }
                        ]
                    },
                    "error": {
                        "code": 0,
                        "key": "string",
                        "message": "string",
                        "details": "string",
                        "request_id": "v-0a885062c80375740f",
                        "resource_id": "rf_0c5d710a87c8a31f86",
                        "resource_type": "voucher"
                    }
                },
                "order": {},
                "validation_rule_id": "string",
                "applicable_to": {
                    "data": [
                        {
                            "object": "product",
                            "id": "string",
                            "source_id": "string",
                            "product_id": "string",
                            "product_source_id": "string",
                            "strict": true,
                            "price": 0,
                            "price_formula": 0,
                            "effect": "APPLY_TO_EVERY",
                            "quantity_limit": 0,
                            "aggregated_quantity_limit": 0,
                            "amount_limit": 0,
                            "aggregated_amount_limit": 0,
                            "order_item_indices": [
                                0
                            ]
                        }
                    ],
                    "total": 0,
                    "object": "list",
                    "data_ref": "data"
                },
                "inapplicable_to": {
                    "data": [
                        {
                            "object": "product",
                            "id": "string",
                            "source_id": "string",
                            "product_id": "string",
                            "product_source_id": "string",
                            "strict": true,
                            "price": 0,
                            "price_formula": 0,
                            "effect": "APPLY_TO_EVERY",
                            "quantity_limit": 0,
                            "aggregated_quantity_limit": 0,
                            "amount_limit": 0,
                            "aggregated_amount_limit": 0,
                            "order_item_indices": [
                                0
                            ]
                        }
                    ],
                    "total": 0,
                    "object": "list",
                    "data_ref": "data"
                },
                "metadata": {},
                "categories": [
                    {
                        "id": "string",
                        "name": "string",
                        "hierarchy": 0,
                        "object": "category",
                        "created_at": "2022-07-14T10:45:13.156Z",
                        "updated_at": "2022-08-16T10:52:08.094Z",
                        "stacking_rules_type": "JOINT"
                    }
                ],
                "banner": "Order Paid - You will get 100 points",
                "name": "promotion_tier_get_points",
                "campaign_name": "PromotionCampaign",
                "campaign_id": "camp_Mow7u4gSxagLlZ2oDQ01ZS5N",
                "validation_rules_assignments": {
                    "object": "list",
                    "data_ref": "data",
                    "data": [
                        {
                            "id": "asgm_LnY1g7UNFA9KyDrD",
                            "rule_id": "val_3gPNA6SnH4ae",
                            "related_object_id": "camp_CZOnEGiZfwIKWmSjhIoIT7Ol",
                            "related_object_type": "campaign",
                            "object": "validation_rules_assignment",
                            "validation_status": "PARTIALLY_VALID",
                            "validation_omitted_rules": [
                                "1"
                            ]
                        }
                    ],
                    "total": 0
                },
                "redeemables": [
                    {
                        "id": "string",
                        "object": "campaign",
                        "created_at": "2022-03-09T11:19:04.819Z",
                        "result": {
                            "discount": {
                                "type": "AMOUNT",
                                "amount_off": 0,
                                "amount_off_formula": "string",
                                "aggregated_amount_limit": 0,
                                "effect": "APPLY_TO_ORDER",
                                "is_dynamic": true
                            },
                            "gift": {
                                "balance": 0,
                                "credits": 0
                            },
                            "loyalty_card": {
                                "points": 7000,
                                "balance": 6970,
                                "exchange_ratio": 0,
                                "points_ratio": 0,
                                "transfers": [
                                    {
                                        "code": "string",
                                        "points": 0,
                                        "reason": "string",
                                        "source_id": "string"
                                    }
                                ]
                            },
                            "error": {
                                "code": 0,
                                "key": "string",
                                "message": "string",
                                "details": "string",
                                "request_id": "v-0a885062c80375740f",
                                "resource_id": "rf_0c5d710a87c8a31f86",
                                "resource_type": "voucher"
                            }
                        },
                        "order": {},
                        "validation_rule_id": "string",
                        "applicable_to": {
                            "data": [
                                {
                                    "object": "product",
                                    "id": "string",
                                    "source_id": "string",
                                    "product_id": "string",
                                    "product_source_id": "string",
                                    "strict": true,
                                    "price": 0,
                                    "price_formula": 0,
                                    "effect": "APPLY_TO_EVERY",
                                    "quantity_limit": 0,
                                    "aggregated_quantity_limit": 0,
                                    "amount_limit": 0,
                                    "aggregated_amount_limit": 0,
                                    "order_item_indices": [
                                        0
                                    ]
                                }
                            ],
                            "total": 0,
                            "object": "list",
                            "data_ref": "data"
                        },
                        "inapplicable_to": {
                            "data": [
                                {
                                    "object": "product",
                                    "id": "string",
                                    "source_id": "string",
                                    "product_id": "string",
                                    "product_source_id": "string",
                                    "strict": true,
                                    "price": 0,
                                    "price_formula": 0,
                                    "effect": "APPLY_TO_EVERY",
                                    "quantity_limit": 0,
                                    "aggregated_quantity_limit": 0,
                                    "amount_limit": 0,
                                    "aggregated_amount_limit": 0,
                                    "order_item_indices": [
                                        0
                                    ]
                                }
                            ],
                            "total": 0,
                            "object": "list",
                            "data_ref": "data"
                        },
                        "metadata": {},
                        "categories": [
                            {
                                "id": "string",
                                "name": "string",
                                "hierarchy": 0,
                                "object": "category",
                                "created_at": "2022-07-14T10:45:13.156Z",
                                "updated_at": "2022-08-16T10:52:08.094Z",
                                "stacking_rules_type": "JOINT"
                            }
                        ],
                        "banner": "Order Paid - You will get 100 points",
                        "name": "promotion_tier_get_points",
                        "campaign_name": "PromotionCampaign",
                        "campaign_id": "camp_Mow7u4gSxagLlZ2oDQ01ZS5N",
                        "validation_rules_assignments": {
                            "object": "list",
                            "data_ref": "data",
                            "data": [
                                {
                                    "id": "asgm_LnY1g7UNFA9KyDrD",
                                    "rule_id": "val_3gPNA6SnH4ae",
                                    "related_object_id": "camp_CZOnEGiZfwIKWmSjhIoIT7Ol",
                                    "related_object_type": "campaign",
                                    "object": "validation_rules_assignment",
                                    "validation_status": "PARTIALLY_VALID",
                                    "validation_omitted_rules": [
                                        "1"
                                    ]
                                }
                            ],
                            "total": 0
                        }
                    }
                ]
            }
        ],
        "total": 5,
        "has_more": true,
        "more_starting_after": "2023-10-31T12:13:16.374Z"
    },
    "tracking_id": "string",
    "order": {},
    "stacking_rules": {
        "redeemables_limit": 30,
        "applicable_redeemables_limit": 5,
        "applicable_exclusive_redeemables_limit": 1,
        "exclusive_categories": [
            "string"
        ],
        "joint_categories": [
            "string"
        ]
    }
}
```

Reply:
I've just check it. You should create promotion with promotion stack and use specific scenario
Payload:
```
{
    "mode": "BASIC",
    "scenario" : "PROMOTION_STACKS"
}
```

Response:
```
{
    "redeemables": {
        "object": "list",
        "data_ref": "data",
        "data": [
            {
                "id": "stack_Zb22FW0F4UAzz9heNaI1XnCj",
                "object": "promotion_stack",
                "created_at": "2023-11-20T09:59:29.360Z",
                "redeemables": [
                    {
                        "id": "promo_DJGQ62yKY3oy3Wkp9C48ShHi",
                        "object": "promotion_tier",
                        "created_at": "2023-11-20T09:59:28.265Z",
                        "result": {
                            "discount": {
                                "type": "AMOUNT",
                                "effect": "APPLY_TO_ORDER",
                                "amount_off": 2000
                            }
                        },
                        "applicable_to": {
                            "data": [],
                            "total": 0,
                            "data_ref": "data",
                            "object": "list"
                        },
                        "inapplicable_to": {
                            "data": [],
                            "total": 0,
                            "data_ref": "data",
                            "object": "list"
                        }
                    }
                ]
            },
            {
                "id": "stack_H9RSuQp6T97jLiSuiCAh2Rbd",
                "object": "promotion_stack",
                "created_at": "2023-11-20T09:59:28.777Z",
                "redeemables": [
                    {
                        "id": "promo_zHXP28kq5lXppzf95likISQJ",
                        "object": "promotion_tier",
                        "created_at": "2023-11-20T09:59:28.054Z",
                        "result": {
                            "discount": {
                                "type": "AMOUNT",
                                "effect": "APPLY_TO_ORDER",
                                "amount_off": 1000
                            }
                        },
                        "applicable_to": {
                            "data": [],
                            "total": 0,
                            "data_ref": "data",
                            "object": "list"
                        },
                        "inapplicable_to": {
                            "data": [],
                            "total": 0,
                            "data_ref": "data",
                            "object": "list"
                        }
                    }
                ]
            }
        ],
        "total": 2,
        "has_more": false
    },
    "stacking_rules": {
        "redeemables_limit": 3,
        "applicable_redeemables_limit": 1,
        "applicable_exclusive_redeemables_limit": 1,
        "exclusive_categories": [],
        "joint_categories": []
    }
}
```